.circleci/config.yml

```diff
@@ -1,140 +1,114 @@
-      gradle-cache-prefix:
-        type: string
-        default: v1
-      build-cache-prefix:
-        type: string
-        default: v1
-
-    # List of available Android Docker images: https://circleci.com/developer/images/image/cimg/android#image-tags
     executor:
-      name: android/android-machine
-      resource-class: large
-      tag: 2022.09.1
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1-node
+    environment:
+      BUILD_FOLDER: code/edgeconsent/build
 
     steps:
       - checkout
 
-      - android/restore-gradle-cache:
-          cache-prefix: << parameters.gradle-cache-prefix >>
-      - android/restore-build-cache:
-          cache-prefix: << parameters.build-cache-prefix >>
-      
-      - run:
-          name: Check Code Format
-          command: make format-check
+      - android/restore-gradle-cache
 
-      - run:
-          name: Lint Code
-          command: make lint
-          
-      - android/save-gradle-cache:
-          cache-prefix: << parameters.gradle-cache-prefix >>
-      - android/save-build-cache:
-          cache-prefix: << parameters.build-cache-prefix >> 
+      - android/restore-build-cache
+
+      # - run:
+      #     name: Check Format
+      #     command: make checkformat
+
+      # - run:
+      #     name: Lint
+      #     command: make lint
+
+      - android/save-gradle-cache
+
+      - android/save-build-cache
 
       # Store Lint report
       - store_artifacts:
-          path: code/edgeconsent/build/reports     
+          path: ${BUILD_FOLDER}/reports     
 
   build-and-unit-test:
-    parameters:
-      gradle-cache-prefix:
-        type: string
-        default: v1
-      build-cache-prefix:
-        type: string
-        default: v1
-
-    # List of available Android Docker images: https://circleci.com/developer/images/image/cimg/android#image-tags
     executor:
-      name: android/android-machine
-      resource-class: large
-      tag: 2022.09.1
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1-node
  
     steps:
       - checkout
 
-      - android/restore-gradle-cache:
-          cache-prefix: << parameters.gradle-cache-prefix >>
-      - android/restore-build-cache:
-          cache-prefix: << parameters.build-cache-prefix >>
+      - android/restore-gradle-cache
+       
+      - android/restore-build-cache
 
       - run:
-          name: Build Library
+          name: Javadoc
+          command: make javadoc
+
+      - store_artifacts:
+          path: ${BUILD_FOLDER}/docs/javadoc
+
+      - run:
+          name: Assemble Phone
           command: make assemble-phone
 
       - run:
-          name: Build Test App
+          name: Assemble for Jitpack
+          command: make ci-publish-maven-local-jitpack
+
+      - run:
+          name: Assemble App
           command: make assemble-app
 
       - run:
           name: Run Unit Tests
           command: make unit-test-coverage
```

Comment:
You can use the Android Orb here to run the unit tests. The main thing it adds is automatic rerun of tests on failure.
```
     - android/run-tests:
          test-command: make unit-test-coverage
```